common/djangoapps/student/views.py

```diff
@@ -2285,31 +2291,65 @@ def auto_auth(request):
 @ensure_csrf_cookie
 def activate_account(request, key):
     """When link in activation e-mail is clicked"""
-    regs = Registration.objects.filter(activation_key=key)
-    if len(regs) == 1:
+
+    # If request is in Studio call the appropriate view
+    if theming_helpers.get_project_root_name().lower() == u'cms':
+        return activate_account_studio(request, key)
+
+    try:
+        registration = Registration.objects.get(activation_key=key)
+    except (Registration.DoesNotExist, Registration.MultipleObjectsReturned):
+        messages.error(
+            request, 'Something went wrong, Make sure you used correct activation url.',
```

Comment:
How about something like, `'Something went wrong, please <a href="{support_url}">contact support</a> to resolve this issue.'`?

Also, we need to mark these messages for translation.